core/deployment/src/main/java/io/quarkus/deployment/dev/RuntimeUpdatesProcessor.java

```diff
@@ -169,19 +185,67 @@ public Path getClassesDir() {
                 .collect(toList());
     }
 
-    private Timer startTestScanningTimer() {
+    private void startTestScanningTimer() {
         synchronized (this) {
-            if (timer == null) {
-                timer = new Timer("Test Compile Timer", true);
-                timer.schedule(new TimerTask() {
-                    @Override
-                    public void run() {
-                        periodicTestCompile();
+            if (testClassChangeWatcher == null && testClassChangeTimer == null) {
+                if (IS_LINUX) {
```

Comment:
I'm almost sure that the `WatchService` also works in Windows and Mac OSX. What's the reason for not using on other OSes?

Reply:
On mac it is just a polling based impl, that polls much slower than our current version (or at least it was when I last used it). Not sure on windows, I need to check, but I thought I would start with Linux as I know it works.